modules/python/pylib/Makefile.am

```diff
@@ -45,27 +45,27 @@ if ENABLE_PYTHON
 PYLIB_PATH = modules/python/pylib
 PYLIB_BUILDDIR = $(abs_builddir)/$(PYLIB_PATH)
 PYLIB_SRCDIR = $(top_srcdir)/modules/python/pylib
-SETUPPY_MANIFEST = $(PYLIB_BUILDDIR)/install-manifest.txt
+PYLIB_MANIFEST = $(PYLIB_BUILDDIR)/install-manifest.txt
 PYTHON_ROOT = $(if $(DESTDIR),$(DESTDIR),/)
 
 INSTALL_EXEC_HOOKS += install-pylib
 UNINSTALL_HOOKS += uninstall-pylib
 CLEAN_HOOKS += clean-pylib
 
-PYSETUP_OPTIONS ?= --root="$(PYTHON_ROOT)" --install-lib="$(python_moduledir)"
+PYLIB_SETUP_OPTIONS ?= --root="$(PYTHON_ROOT)" --install-lib="$(python_moduledir)"
```

Comment:
In case we want to install 3rd party dependencies too into our own python folder, the following options may come in handy to workaround a few pip bugs around custom install dirs (they sometimes mix up system dirs with custom dirs when checking which deps to install):

```
--ignore-installed --cache-dir "$(PYLIB_BUILDDIR)/cache" --build "$(PYLIB_BUILDDIR)/build"
```

Reply:
hmm.. this variable is now unused as it seems. when I migrated to using pip to install packages. I am not sure when you did this review. 

at the moment I have --upgrade so it overwrites the existing installation (which --ignore-installed is probably achieving)

customizing build/cache directories might make sense, let me add those... 

Hmm. as it seems neither build or cache are supported by pip, so I guess those options only apply to setup.py

I am not sure they are needed now, though. I was reading packaging manuals for hours to come up with the pip based solution as it seems to be the way forward now. Using setup.py to install packages directly is not supported anymore.

Reply:
`--ignore-installed` was necessary for our internal modules to avoid a few known pip install prefix bugs.
I'll find and reference those issues soon.

Reply:
nevermind, I am still fighting with setuptools/pip and related problems. I previously wanted to migrate to using pip (due to setuptools claiming our use-case was deprecated) but I am now thinking about adopting poetry instead. That seems to be much better. 

But instead of  rolling that into this PR (already too big), I now try to stick to setup.py only, as it worked for us until now, I am not trying to find the replacement features in pip (if they exist at all).

I am also thinking about merging modules/python/pylib and modules/python-modules to install a single "syslogng" package (both the code exporting core features, and debuggercli and any official python modules).

But let me first try to make the current state work, with setup.py and then see if the merger of these modules would help.

Reply:
I think with the virtualenv management stuff, this is resolved.